kod/object/active/holder/nomoveon/battler/monster/towns/marntwn/mrinnk.kod

```diff
@@ -107,6 +110,39 @@ messages:
             return;
          }
 
+         if (StringContain(string, MrIk_room_command) AND StringContain(string, MrIk_help_command))
```

Comment:
I haven't read through this whole change yet, but definitely repeating the exact same code in 3 different places (the 3 innkeepers) is a big no-no.

Reply:
I agree and imo the entire room system is pretty gross under the hood, but that's how it was done for the original commands that are already in place.   I could set up a function inside the room maintenance object that each innkeeper would call to check for and handle these new player-issued commands in one place instead, but the originals would still be there in their current messy form.

Reply:
After thinking about this a bit more, the following might be the best approach, which will mean I close this PR and break it up into several smaller chunks:

- Move the existing duplicated code in SomeoneSaid from the 3 innkeepers into monster.kod for a proper subclass/superclass relationship.
- Add a new constant for viAttributes (MOB_LANDLORD) which must be set to enable room rental behavior (similar to how MOB_SELLER, MOB_BANKER, etc works).
- Apply MOB_LANDLORD to existing npcs that rent rooms (morrigan, pritchett, moxal)
- Add the copy key drop protection as PR 1
- Add the new commands as PR 2, maybe even just 1 new command per PR

Thoughts on this?